lte/gateway/c/core/oai/lib/openflow/controller/GTPApplication.cpp

```diff
@@ -541,21 +545,37 @@ void GTPApplication::add_downlink_arp_flow_action(
   downlink_fm.add_instruction(goto_inst);
 
   // Finally, send flow mod
-  messenger.send_of_msg(downlink_fm, ev.get_connection());
+  messenger.send_of_msg(downlink_fm, conn);
   OAILOG_DEBUG_UE(LOG_GTPV1U, imsi, "ARP flow added\n");
 }
 
-void GTPApplication::add_downlink_arp_flow(
-    const AddGTPTunnelEvent& ev, const OpenflowMessenger& messenger,
+void GTPApplication::add_downlink_arp_flow_(
+    fluid_base::OFConnection* conn, int precedence, const std::string imsi_,
+    const struct in_addr& ue_ip, const OpenflowMessenger& messenger,
     uint32_t ingress_port) {
-  uint32_t flow_priority =
-      convert_precedence_to_priority(ev.get_dl_flow_precedence());
+  uint32_t flow_priority = convert_precedence_to_priority(precedence);
   of13::FlowMod downlink_fm =
       messenger.create_default_flow_mod(0, of13::OFPFC_ADD, flow_priority);
 
-  add_downlink_arp_match(downlink_fm, ev.get_ue_ip(), ingress_port);
+  add_downlink_arp_match(downlink_fm, ue_ip, ingress_port);
 
-  add_downlink_arp_flow_action(ev, messenger, downlink_fm);
+  add_downlink_arp_flow_action(conn, imsi_, messenger, downlink_fm);
+}
+
+void GTPApplication::add_downlink_arp_flow(
+    const AddGTPTunnelEvent& ev, const OpenflowMessenger& messenger,
+    uint32_t ingress_port) {
+  add_downlink_arp_flow_(
+      ev.get_connection(), ev.get_dl_flow_precedence(), ev.get_imsi(),
+      ev.get_ue_ip(), messenger, ingress_port);
+}
+
+void GTPApplication::add_downlink_arp_flow_paging_event(
```

Comment:
This function implementation is exactly same as `add_downlink_arp_flow`. Do we still need it ?

Reply:
add_downlink_arp_flow () is processing AddGTPTunnelEvent, this function is processing AddArpFlowEvent event.

Reply:
Got it, thanks for the clarification!